pkg/operator/ceph/config/config.go

```diff
@@ -138,6 +138,17 @@ func SetOrRemoveDefaultConfigs(
 		}
 	}
 
+	// remove `rbd_default_features` since it is no longer used in the
+	// upgraded clusters. This was set to `3` previously from Rook.
+	rbdDefaultFeatures := "rbd_default_features"
+	val, err := monStore.Get("mon", rbdDefaultFeatures)
+	if err == nil && val == "3" {
+		logger.Infof("removing rbd_default_features config")
+		err = monStore.Delete("global", rbdDefaultFeatures)
```

Comment:
This is idempotent IIRC so it's probably not worth doing a Get then a Delete.

Reply:
@leseb This is done to make sure not to delete the configuration manually set by the user (other than `3`)

Reply:
Oh I see, yes, sounds good then, where do you handle the error on the `Get`?

Reply:
Not worried about the get error. Do I need to consider errors and return error other than non-exist?

Reply:
added a check now, This should not fail. it should get the default values that ceph provides.

Reply:
I Dont have an answer for this one. Any values other than `3` will be kept as is it. Only 3 will be reverted back. Is it possible to handle this case?

Reply:
Maybe they can set it in the ceph.conf and use it?

Reply:
> We have no idea, I'm not sure about the impact if someone sets 3 and really wants it and we remove it each time we reconcile...

I am concerned about this also. We don't want to be constantly clobbering this value if users want to set it. And if this is the only setting that we require users to set in the config override, that starts to become hard to document and hard for users to understand. However, I do think it's reasonable to clobber the value once during upgrade, with a note in the upgrade docs for users to pay attention.

We can heuristically determine if we are currently processing a Rook upgrade, though we don't have any code currently to do that. There is a `rook-version` label on all deployments that will have some text like `rook-version=v1.8.6` (possibly with some trailing info like `rook-version=v1.8.0-alpha.0-336-gb717f4d06-dirty`) that tells the Rook version that configured the deployment. We could parse the `rook-version` label from all the mon deployments and determine the lowest-versioned pod. Then, we would un-set `rbd_default_features` if (and only if) the Rook X or Y (`vX.Y.Z`) version for the current binary is greater than the lowest `rook-version` label on a mon pod. We can include a note in the release notes and upgrade guide that if users wish to set that value themselves, they should do so (again) after the upgrade. We should be sure to remove that upgrade process logic before v1.10 so we don't clobber it again when upgrading from v1.9 to v1.10.

Thoughts @travisn ?

Reply:
@Madhu-1 To be clear, what is the impact if we don't merge this? What is the impact if the setting still remains after upgrade? Do upstream users care? 

If upstream users care, we could just put a note in the upstream guide to document that they may want to delete this setting from the toolbox. 

If this is really only needed downstream, we could just make this change in the 4.10 release branch and no need to keep it in an upstream branch.

Reply:
FWIW we are fortunate that Rook picked `3` here.  In my experience, few users touch `rbd_default_features` and those that do wouldn't set it to `3` because it is a bit meaningless (`3` stands for `layering,striping` and `striping` is an internal feature bit over which users have no direct control).  A manual setting by a user who did it on purpose and actually cared would likely be either `1` or something bigger than `3`.

So from an RBD perspective, I'm fine with this bit and retract my "hypothetically, a user could have manually set ..." concern from an earlier comment.  I do want the `monStore.Get` bit to be addressed though.

Reply:
Whether it's worth it upstream or should only be done downstream is a separate discussion ;)

Reply:
For upstream I'd propose we don't make this code change and just add a note to the upgrade guide.

Reply:
Late of the party :) Am fine with documentation for the upstream and will send a downstream-only patch to remove this setting.